File: 38-search-a-2d-matrix-ii/standard_answer.cc
class Solution {
public:
    /**
     * @param matrix: A list of lists of integers
     * @param target: An integer you want to search in matrix
     * @return: An integer indicate the total occurrence of target in the given matrix
     */
    int searchMatrix(vector<vector<int>> &matrix, int target) {
        // write your code here
        if (matrix.empty() || matrix[0].empty()) {
            return 0;
        }
        int n = matrix.size();
        int m = matrix[0].size();
        int x = n - 1;
        int y = 0;
        int count = 0;
        while (x >= 0 && y < m) {
            if (matrix[x][y] < target) {
                y++;
            } else if (matrix[x][y] > target) {
                x--;
            } else {
                count++;
                y++;
                x--;
            }
        }
        return count;
    }
};

File: 196-missing-number/standard_answer.java
public class Solution {
    /**
     * @param nums: An array of integers
     * @return: An integer
     */
    private int MissingNum = -1;
    public int findMissing(int[] nums) {
        // write your code here
        int N = nums.length;
        int sum1 = 0;
        for (int elem : nums) {
            sum1 += elem;
        }
        return (N * (N + 1) / 2) - sum1;
    }
}
